### PR TITLE
Iss728 high_time_resolution to time_resolved - Phase 1

### LINKED ISSUE #728: Update "high_time_resolution" naming to be "time_resolved"
The current use of "high_time_resolution" in relation to footprint metadata is misleading. The terminology "time_resolved" would be more appropriate.

To update this we would likely need to take a phased approach:
1. Add an additional keyword for "time_resolved", make sure if "high_time_resolution" the details are stored within "time_resolved" instead and raise a DepreceationWarning
2. Allow for backwards compatibility but remove ability to pass this keyword at the front end

This is relevant to standardise_footprint and Footprint.read_file inputs.

### PR DESCRIPTION
**Pull request by SutarPrasad**
* **Summary of changes** (Bug fix, feature, docs update, ...)

- [x] Added a `time_resolved` as an argument wherever there is an occurrence of `high_time_resolution`
- [x] Added deprecation warning for `high_time_resolution`
- [x] Changed the metadata dictionary key from `high_time_resolution` to `time_resolved`
- [x] Made the necessary changes to the tests. At some instances kept `high_time_resolution` as a passing argument in tests to verify the scenario of value ultimately gets stored in time_resolved.

* **Please check if the PR fulfills these requirements**

- [x] Closes #728  (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
- [x] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors
- [x] Documentation and tutorials updated/added
- [x] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature
- [x] Added any new requirements to `requirements.txt` and `recipes/meta.yaml`

### CODE REVIEW

File: openghg/standardise/_standardise.py [hunk 1]
```diff
@@ -487,6 +487,7 @@ def standardise_footprint(
     overwrite: bool = False,
     force: bool = False,
     high_spatial_resolution: bool = False,
+    time_resolved: Optional[bool] = False,
```
Comment (gareth-j) [inline openghg/standardise/_standardise.py:490]:
```suggestion
    time_resolved: bool = False,
```

File: openghg/standardise/_standardise.py [hunk 2]
```diff
@@ -620,6 +629,7 @@ def standardise_flux(
     source_format: str = "openghg",
     database_version: Optional[str] = None,
     model: Optional[str] = None,
+    time_resolved: Optional[bool] = False,
```
Comment (gareth-j) [inline openghg/standardise/_standardise.py:632]: We shouldn't use `Optional` here as we always expect them to be bools.

File: openghg/standardise/_standardise.py [hunk 3]
```diff
@@ -620,6 +629,7 @@ def standardise_flux(
...
     high_time_resolution: Optional[bool] = False,
```
Comment (gareth-j) [inline openghg/standardise/_standardise.py:633]: Update these as well.

File: openghg/standardise/flux/_intem.py [hunk 4]
```diff
@@ -11,6 +12,7 @@ def parse_intem(
     domain: str = "europe",
     model: str = "intem",
     period: Optional[Union[str, tuple]] = None,
+    time_resolved: Optional[bool] = False,
```
Comment (gareth-j) [inline openghg/standardise/flux/_intem.py:15]: Remove Optional

File: openghg/standardise/flux/_openghg.py [hunk 5]
```diff
@@ -11,6 +12,7 @@ def parse_openghg(
     database: Optional[str] = None,
     database_version: Optional[str] = None,
     model: Optional[str] = None,
+    time_resolved: Optional[bool] = False,
```
Comment (gareth-j) [inline openghg/standardise/flux/_openghg.py:15]: Remove optional.

File: openghg/store/_flux.py [hunk 6]
```diff
@@ -64,6 +65,7 @@ def read_file(
         database_version: Optional[str] = None,
         model: Optional[str] = None,
         source_format: str = "openghg",
+        time_resolved: Optional[bool] = False,
```
Comment (gareth-j) [inline openghg/store/_flux.py:68]: Remove optional

**Review by gareth-j — CHANGES_REQUESTED** [review]
Some changes needed to types.

File: openghg/standardise/_standardise.py (hunk 1 above)
Comment (SutarPrasad) [reply openghg/standardise/_standardise.py]: done

**Review by gareth-j — APPROVED** [review]
Looks good.

**Pull request merged**